src/servers/Server.cc

```diff
@@ -206,14 +203,20 @@ Server::clientWriteDone(const CommIoCbParams &io)
 
     if (io.flag) {
         debugs(33, 2, "bailing after a write failure: " << xstrerr(io.xerrno));
-        LogTagsErrors lte;
-        lte.timedout = io.xerrno == ETIMEDOUT;
-        lte.aborted = !lte.timedout; // intentionally true for zero io.xerrno
-        terminateAll(Error(ERR_WRITE_ERROR, SysErrorDetail::NewIfAny(io.xerrno)), lte);
+        terminateWithError(ERR_WRITE_ERROR, io.xerrno);
```

Comment:
I would like to avoid adding a second terminateX() method just to encapsulate Error and LogTagsErrors creation code, especially when that encapsulation moves us back from a flexible/safe/explicit ErrorDetail API to none-of-those-things int-is-errno API. Adding terminateX() may also encourage creation of terminateY() and terminateZ() to accommodate other error detail construction cases, and we do not want that kind of API fan-out.

I realize that part of the appeal here is how LogTagsErrors objects are initialized from errno. I am pretty sure that code should eventually disappear because we can compute the same log tags based on the information stored in Error. However, polishing that aspect will result in more changes than I would like to see in this particular PR.

Instead of adding terminateWithError(), please add and use a LogTagsErrors::FromErrno() static method that returns a properly configured LogTagsErrors object. Doing so will arguably leave some of the logic duplicated, but I think it is better than adding a competing terminateX() interface with raw errno argument.

```suggestion
        terminateAll(Error(ERR_WRITE_ERROR, SysErrorDetail::NewIfAny(io.xerrno)), LogTagsErrors::FromErrno(io.xerrno));
```

Reply:
Done (5a03963).